runtime/shared_common/shrinit.cpp

```diff
@@ -2624,6 +2625,44 @@ performSharedClassesCommandLineAction(J9JavaVM* vm, J9SharedClassConfig* sharedC
 	return J9VMDLLMAIN_SILENT_EXIT_VM;
 }
 
+/* A helper to retrieve user name and store it into the buffer provided by the caller.
+ *
+ * @param[in] vm The Java VM
+ * @param[out] buffer Buffer for the user name
+ * @param[in] length The length of the buffer
+ *
+ * @return 0 on success, number of bytes required to hold the user name
+ * if the output buffer was too small, -1 on failure.
+ */
+static IDATA
+sysinfoGetUserNameHelper(J9JavaVM *vm, UDATA verboseFlags, char *buffer, UDATA length)
+{
+	PORT_ACCESS_FROM_JAVAVM(vm);
+	IDATA rc = -1;
+
+#if defined(J9VM_OPT_CRIU_SUPPORT)
+	/* Skip j9sysinfo_get_username if a checkpoint can be taken. */
+	J9InternalVMFunctions *vmFuncs = vm->internalVMFunctions;
+	if (!vmFuncs->isCheckpointAllowed(vmFuncs->currentVMThread(vm)))
+#endif /* defined(J9VM_OPT_CRIU_SUPPORT) */
+	{
+		rc = j9sysinfo_get_username(buffer, length);
+		if (rc > 0) {
+			SHRINIT_ERR_TRACE2(verboseFlags, J9NLS_SHRC_SHRINIT_FAILURE_COPYING_USERNAME_TOOLONG, length, rc);
+		}
+	}
+	if (rc < 0) {
+		rc = j9sysinfo_get_env("USER", buffer, length);
```

Comment:
For the user name, we are trying `getpwuid()` first, then the environment variable, which is inconsistent with `j9shmem_getDir()` where the environment variable is queried first.

Reply:
Switched the order within `sysinfoGetUserNameHelper()`, attempt `j9sysinfo_get_env()` first.